src/vibrato.rs

```diff
@@ -0,0 +1,143 @@
+// create a vibrato processor that is able to add vibrato effect to each block of audio data that works with multiple channels.
+
+use crate::ring_buffer::RingBuffer;
+
+use crate::lfo::LFO; 
+
+/// Vibrato is a struct that contains a delay line, an LFO, and a width, sample rate, and delay.
+pub struct Vibrato {
+    delay_line: Vec<RingBuffer<f32>>,
+    lfo: LFO,
+    width: f32,
+    sample_rate: f32,
+    delay: f32,
+}
+
+impl Vibrato {
+    /// Create a new vibrato processor with a given sample rate, maximum delay, delay, width, frequency, and number of channels.
+    pub fn new(sample_rate: f32, max_delay: f32, delay: f32, width: f32, frequency: f32, channels: usize) -> Vibrato {
```

Comment:
I'd recommend adding RustDocs talking about your params. Here's how I did on my new():
    /// Constructs a new Vibrato struct.
    /// * `sr` - The Sample Rate of the Vibrato
    /// * `hz` - The frequency of the Vibrato
    /// * `depth` - The depth of the Vibrato
    /// * `sec` - The time of the Vibrato

The * `sec` formats it to RustDocs var stuff